lib/rubocop/cop/layout/tab.rb

```diff
@@ -52,21 +53,31 @@ def autocorrect(range)
 
         private
 
-        def string_literal_lines(ast)
+        # rubocop:disable Metrics/CyclomaticComplexity
+        def in_string_literal?(ranges, line, col)
+          ranges.any? do |range|
+            (range.line == line && range.column <= col) ||
+              (range.line < line && line < range.last_line) ||
+              (range.line != line && range.last_line == line &&
+               range.last_column >= col)
+          end
+        end
+        # rubocop:enable Metrics/CyclomaticComplexity
+
+        def string_literal_ranges(ast)
           # which lines start inside a string literal?
           return [] if ast.nil?
 
-          ast.each_node(:str, :dstr).each_with_object(Set.new) do |str, lines|
+          ast.each_node(:str, :dstr).each_with_object(Set.new) do |str, ranges|
             loc = str.location
 
-            str_lines = if loc.is_a?(Parser::Source::Map::Heredoc)
-                          body = loc.heredoc_body
-                          (body.first_line)..(body.last_line)
-                        else
-                          (loc.first_line + 1)..(loc.last_line)
-                        end
+            range = if loc.is_a?(Parser::Source::Map::Heredoc)
```

Comment:
I think there's a `StrNode#heredoc?` now, but not 100% sure.

Reply:
Definitely. I forgot the helper. Thanks!